smart_campus/app/views.py

```diff
@@ -992,29 +997,33 @@ def travelplan_edit_page(request, pk):
 
         if travelplan_form.is_valid():
             edited_travelplan = travelplan_form.save()
-
             json_order = json.loads(request.POST['order'])
-
-            if not json_order:
-                for travelplan_station in TravelPlanStations.objects.filter(travelplan_id=pk):
-                    travelplan_station.delete()
-
-            else:
-                for order, station_id in enumerate(json_order):
-                    changed_travelplan = TravelPlanStations.objects.filter(
+            exist_travelplan_stations = []
+            for travelplan_station in TravelPlanStations.objects.filter(travelplan_id=pk):
+                exist_travelplan_stations.append(travelplan_station.station_id)
```

Comment:
You can write line 1001~1003 as a list comprehension.
List comprehension is slightly faster.

Reply:
Got it.